File: app/Http/Controllers/AuthenticationController.php
<?php

namespace App\Http\Controllers;

use App\Models\User;
use Illuminate\Http\Request;
use Illuminate\Http\Response;
use Illuminate\Support\Facades\Hash;

class AuthenticationController extends Controller {
  public function getUser() {
    return response()->json([
      "user" => auth()->user()
    ], Response::HTTP_OK);
  }
  
  public function login(Request $request) {
    $validated = $request->validate([
      "email" => "string|required",
      "password" => "string|required"
    ]);

    $email = $username = $validated['email'];

    $user = User::where("email", $email)->orWhere("username", $username)->first();

    if (!$user || !Hash::check($validated['password'], $user->password)) {
      return response()->json([
        "message" => "Bad Credentials."
      ], Response::HTTP_OK);
    }

    $token = $user->createToken("login_token")->plainTextToken;
 
    return response()->json([
      'token' => $token,
      "user" => $user,
    ], Response::HTTP_OK);
  }

  public function register() {
  }

  public function logout() {
    auth()->user()->tokens()->delete();

    return response()->json([
      'message' => 'Successfully logged out.'
    ], Response::HTTP_OK);
  }
}

File: app/app/src/main/java/umn/ac/id/maggotproject/controller/AuthenticationController.java
package umn.ac.id.maggotproject.controller;

import android.util.Log;

import umn.ac.id.maggotproject.model.UserModel;

public class AuthenticationController {
    private String token;
    private String message;
    private UserModel.User user;

    public Result login() {
        return new Result(token, message, user);
    }

    public class Result {
        private String token;
        private UserModel.User user;
        private String message;

        public String getToken() {
            return token;
        }

        public UserModel.User getUser() {
            return user;
        }

        public String getMessage() {
            return message;
        }

        public Result(String token, String message, UserModel.User user) {
            this.token = token;
            this.message = message;
            this.user = user;
        }

        @Override
        public String toString() {
            return "Result{" +
                    "token='" + token + '\'' +
                    ", user=" + user +
                    ", message='" + message + '\'' +
                    '}';
        }
    }
}
